inference-engine/src/gna_plugin/optimizer/gna_pass_manager.cpp

```diff
@@ -319,24 +334,52 @@ void SubstituteSoftSignPass::run() {
         }
         if (cont) continue;
         if (!hasNChildren(abs, 1)) continue;
-        auto power = getNthChild(abs, 0);
+        auto addition = getNthChild(abs, 0);
+        InferenceEngine::CNNLayerPtr power = nullptr;
+
+        if (!LayerInfo(addition).isPower()) continue;
+        auto powerLayer = LayerInfo(addition).as<PowerLayer*>();
+
+        if (!fp32eq(powerLayer->power, -1.0f)) {
+            //if first one is power equals 1, then it still can be a softsign
+            //with two power operations, one for add and one for power
+            if (!fp32eq(powerLayer->power, 1.0f) ||
+                !fp32eq(powerLayer->offset, 1.0f) ||
+                !fp32eq(powerLayer->scale, 1.0f)) {
+                continue;
+            } else {
+                power = getNthChild(addition, 0);
+                if (!LayerInfo(power).isPower()) continue;
+                auto powerLayer_1 = LayerInfo(power).as<PowerLayer*>();
+                if (!fp32eq(powerLayer_1->power, -1.0f) ||
+                    !fp32eq(powerLayer_1->offset, 0.0f) ||
+                        !fp32eq(powerLayer_1->scale, 1.0f)) continue;
+            }
+        } else {
+            if (!fp32eq(powerLayer->offset, 1) ||
+                !fp32eq(powerLayer->scale, 1)) continue;
 
-        if (!LayerInfo(power).isPower()) continue;
-        auto powerLayer = LayerInfo(power).as<PowerLayer*>();
-        if (powerLayer->power != -1) continue;
-        if (powerLayer->offset != 1) continue;
-        if (powerLayer->scale != 1) continue;
+            std::swap(addition, power);
+        }
 
         if (!hasNChildren(power, 1)) continue;
         auto mulSame = getNthChild(power, 0);
         if (mulSame != mul) continue;
 
         // pattern matched - lets substitute
-        gnalog() << "SoftSign subgraph found consits of: \n"
-                 << "\t" << abs->name << "\n"
-                 << "\t" << power->name << "\n"
-                 << "\t" << mul->name << "\n"
-                 << std::endl;
+        if (addition == nullptr)
+            gnalog() << "SoftSign subgraph found consits of: \n"
+                     << "\t" << abs->name << "\n"
+                     << "\t" << power->name << "\n"
+                     << "\t" << mul->name << "\n"
+                     << std::endl;
+        else
+            gnalog() << "SoftSign subgraph found consits of: \n"
+                     << "\t" << abs->name << "\n"
+                     << "\t" << addition->name << "\n"
+                     << "\t" << power->name << "\n"
+                     << "\t" << mul->name << "\n"
+                     << std::endl;
```

Comment:
a lot of code repetition why not
```
gnalog() << "SoftSign subgraph found consits of: \n"
gnalog() << "\t" << abs->name << "\n"
if (addition != nullptr) gnalog() << "\t" << addition->name << "\n"
gnalog() << "\t" << power->name << "\n"
gnalog() << "\t" << mul->name << "\n"
gnalog() << std::endl;
```
Change is not obligatory though, seems like personal preference of mine